linkerd2-proxy/src/main.rs

```diff
@@ -10,6 +10,10 @@ compile_error!(
     "at least one of the following TLS implementations must be enabled: 'meshtls-boring', 'meshtls-rustls'"
 );
 
+// Emit a compile-time error if FIPS mode is enabled but meshtls-boring is not
+#[cfg(all(not(feature = "meshtls-boring"), all(feature = "meshtls-fips")))]
+compile_error!("'meshtls-boring' needs to be enabled to support FIPS mode");
```

Comment:
it seems like an alternative solution would just be to have `meshtls-fips` enable the `meshtls-boring` feature?

the question is whether we would want the `meshtls-fips` feature flag to eventually be able to enable FIPS support with other (non-BoringSSL) TLS implementations (in which case, `meshtls-fips` should not enable the `meshtls-boring` crate), or whether it's specifically configuring the BoringSSL TLS implementation...